### PR TITLE
Update for TyCtxt<'a, 'gcx, 'tcx> -> TyCtxt<'tcx>.

### PR DESCRIPTION
**Pull request by eddyb**
See https://github.com/rust-lang/rust/pull/61722 and https://github.com/rust-lang/rust/pull/61817 for some backstory.

cc @mark-i-m @rust-lang/compiler

### CODE REVIEW

File: src/appendix/glossary.md [hunk 1]
```diff
@@ -25,7 +25,6 @@ early-bound lifetime    |  a lifetime region that is substituted at its definiti
 empty type              |  see "uninhabited type".
 Fat pointer             |  a two word value carrying the address of some value, along with some further information necessary to put the value to use. Rust includes two kinds of "fat pointers": references to slices, and trait objects. A reference to a slice carries the starting address of the slice and its length. A trait object carries a value's address and a pointer to the trait's implementation appropriate to that value. "Fat pointers" are also known as "wide pointers", and "double pointers".
 free variable           |  a "free variable" is one that is not bound within an expression or term; see [the background chapter for more](./background.html#free-vs-bound)
-'gcx                    |  the lifetime of the global arena ([see more](../ty.html))
```
Comment (Zoxc) [inline src/appendix/glossary.md]: This description should be moved to `'tcx`.

File: src/conventions.md [hunk 2]
```diff
@@ -141,8 +141,7 @@ to the compiler.
 - `cx` tends to be short for "context" and is often used as a suffix. For
   example, `tcx` is a common name for the [Typing Context][tcx].
 
-- [`'tcx` and `'gcx`][tcx] are used as the lifetime names for the Typing
-  Context.
+- [`'tcx`][tcx] is used as the lifetim names for the Typing Context.
```
Comment (Zoxc) [inline src/conventions.md]: `lifetim` typo.

Comment (mark-i-m) [reply src/conventions.md]:
```suggestion
- [`'tcx`][tcx] is used as the lifetime names for the Typing Context.
```

File: src/appendix/glossary.md (hunk 1 above)
Comment (mark-i-m) [reply src/appendix/glossary.md]:
```suggestion
'tcx                    |  the lifetime of the allocation arena ([see more](../ty.html))
```

**Comment by eddyb** [comment]
@mark-i-m The `QueryCx` stuff is longer term, now that we got to `tcx: TyCtxt<'tcx>` directly.

**Pull request merged**